exporter/sumologicexporter/sender.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicexporter
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type sender struct {
+	buffer []pdata.LogRecord
+	config *Config
+	client *http.Client
+	filter *filtering
+}
+
+func newSender(cfg *Config, cl *http.Client, f *filtering) *sender {
+	return &sender{
+		config: cfg,
+		client: cl,
+		filter: f,
+	}
+}
+
+// Send sends data to sumologic
+func (s *sender) send(pipeline PipelineType, body *strings.Reader, fields FieldsType) error {
```

Comment:
Accept`io.Reader` - more relaxed rules on what can be passed in here

```suggestion
func (s *sender) send(pipeline PipelineType, body io.Reader, fields FieldsType) error {
```